linkedlistutil.cpp

```diff
@@ -0,0 +1,173 @@
+#include <iostream>
+#include <cstdlib>
+#include <stack>
+#include <cmath>
+#include <vector>
+#include <cstring>
+using namespace std;
+
+
+
+typedef struct node
+{
+	int data;
+	struct node * next;
+	node(int x) {
+		data = x;
+		next = NULL;
+	}
+} node;
+
+node* isJoinedhelper( node* head1, node* head2 )
+  {
+    node * temp = head1;
+    int count1=0,count2=0;
+    while(temp != NULL && temp -> next != NULL)
+    {
+      count1++;
+      temp = temp->next;
+    }
+    node* tail1 = temp;
+    temp = head2;
+    while( temp !=NULL && temp -> next != NULL )
+    {
+      count2++;
+      temp = temp -> next;
+    }
+    node* tail2 = temp;
+    if ( tail1 != tail2 )
+    { 
+        cout << "Not Merging" ;
+        return (tail1 -> next);
+    }
+    else
+    {
+       if ( count1 < count2 )
+       {
+           for ( int i = 0; i < (count2-count1) ; i++)
+               head2 = head2 -> next;
+       }
+      if ( count1 > count2 )
+      {
+           for ( int i = 0; i < (count1-count2) ; i++)
+               head1 = head1 -> next;
+      }
+      while( head1 -> next != head2 -> next)
+        {
+            head1 = head1 -> next;
+            head2 = head2 -> next;
+        }
+      return head1 -> next;
+    }
+}
+
+bool isJoined(node* head1,node*head2)
```

Comment:
good but a bit of spacing issue. need to work on consistent spacing. like
<code> node* head1, node* head2 </code>